### PR TITLE
feat(content): Add Modals and TextInputs.

### LINKED ISSUE #28: Add page for Modals and TextInputs
### Expands on

_No response_

### Title

Modals and Text Inputs

### Description

As modals will be added to the library in https://github.com/DisnakeDev/disnake/pull/253, it's a good idea to cover this by the time the PR is merged - in order to clarify how the feature will be used.

We're on a time limit, gotta do this fast. :P

### Additional context

_No response_

### Would you like to implement this yourself?

Yes, I will be contributing this addition.

### PR DESCRIPTION
**Pull request by abhigyantrips**
## Relevant Issues

Closes #28

### CODE REVIEW

File: code-samples/interactions/modals/create_tag_low.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            ],
+        )
+
+        modal_inter: disnake.ModalInteraction = await self.bot.wait_for(
+            "modal_submit",
+            check=lambda i: i.custom_id == "create_tag_low" and i.author.id == inter.author.id,
+        )
+
+        embed = disnake.Embed(title="Tag Creation")
+        for key, value in modal_inter.values.items():
```
Comment (Victorsitou) [inline code-samples/interactions/modals/create_tag_low.py:39]:
```suggestion
        for key, value in modal_inter.text_values.items():
```

File: code-samples/interactions/modals/create_tag.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                placeholder="Lorem ipsum dolor sit amet.",
+                custom_id="description",
+                style=TextInputStyle.paragraph,
+            ),
+        ]
+        super().__init__(title="Create Tag", custom_id="create_tag", components=components)
+
+    async def callback(self, inter: disnake.ModalInteraction) -> None:
+        embed = disnake.Embed(title="Tag Creation")
+        for key, value in inter.values.items():
```
Comment (Victorsitou) [inline code-samples/interactions/modals/create_tag.py:27]:
```suggestion
        for key, value in inter.text_values.items():
```

**Comment by abhigyantrips** [comment]
Alright, will make this and the `value[:1024]` change right now.

File: guide/docs/interactions/modals.mdx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+# The cog setup function. (Irrelevant)
+def setup(bot: commands.Bot):
+    bot.add_cog(Modals(bot))
+```
+
+## Creating a modal
+
+A modal can be created and sent using two different methods, just like other message components:
+
+-   Subclassing <DocsLink reference="disnake.ui.Modal">disnake.ui.Modal</DocsLink> to define the components, and sending it through the `modal` attribute of <DocsLink reference="disnake.InteractionResponse.send_modal">interaction.send_modal</DocsLink> (as done in the example above).
```
Comment (shiftinv) [inline guide/docs/interactions/modals.mdx:102]:
```suggestion
-   Subclassing <DocsLink reference="disnake.ui.Modal">disnake.ui.Modal</DocsLink> to define the components, and sending it through the `modal` parameter of <DocsLink reference="disnake.InteractionResponse.send_modal">interaction.send_modal</DocsLink> (as done in the example above).
```

File: guide/docs/interactions/modals.mdx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+## `TextInput` components
+
+`TextInput` is a component in itself, like buttons and select menus. It is responsible for receiving user input in the form of text, and cannot be used outside of modals. It has the following customizable attributes.
+
+-   `label`: The label of the text input.
+-   `custom_id`: An ID specified to the text input.
+-   `style`: Utilizes one of <DocsLink reference="disnake.TextInputStyle" /> for the component.
+-   `placeholder`: The placeholder text that is shown if nothing is entered.
+-   `value`: The pre-filled value of the text input (upto character limit).
```
Comment (shiftinv) [inline guide/docs/interactions/modals.mdx:149]:
```suggestion
-   `value`: The pre-filled value of the text input (up to `max_length`).
```

File: guide/docs/interactions/modals.mdx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+### <DocsLink reference="disnake.ui.Modal.callback">`callback`</DocsLink>
+
+### <DocsLink reference="disnake.ui.Modal.on_error">`on_error`</DocsLink>
+
+### <DocsLink reference="disnake.ui.Modal.on_timeout">`on_timeout`</DocsLink>
+
+## Modal notes
+
+-   The <DocsLink reference="disnake.TextInputStyle" /> defaults to `short`, if not specified for the component.
+-   All <DocsLink reference="disnake.ui.TextInput">TextInput</DocsLink> components are required by default.
```
Comment (shiftinv) [inline guide/docs/interactions/modals.mdx:183]: The latter is already mentioned above, and the former could be as well (see review)

File: guide/docs/interactions/modals.mdx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+## Modal limits
+
+-   A modal can have a total of **5 components**.
```
Comment (shiftinv) [inline guide/docs/interactions/modals.mdx:187]:
```suggestion
-   A modal can have a total of **5 action rows**.
```
Also already mentioned above, but having all the limits in a separate section as well is a nice idea.

**Review by shiftinv — COMMENTED** [review]
Pretty cool :o

Examples shouldn't use cogs, just unnecessarily complicates them in my opinion.

File: code-samples/interactions/modals/create-tag-low.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    placeholder="Lorem ipsum dolor sit amet.",
+                    custom_id="description",
+                    style=TextInputStyle.paragraph,
+                ),
+            ],
+        )
+
+        modal_inter: disnake.ModalInteraction = await self.bot.wait_for(
+            "modal_submit",
+            check=lambda i: i.custom_id == "create_tag_low" and i.author.id == inter.author.id,
```
[the remark this answers is not in the document]
Comment (abhigyantrips) [reply code-samples/interactions/modals/create-tag-low.py:35]: Alrighty. Will update both examples to mock the main bot file as well.

File: guide/docs/interactions/modals.mdx (hunk 5 above)
Comment (abhigyantrips) [reply guide/docs/interactions/modals.mdx:183]: This list is supposed to be a summary of all the notes/limits, to act as a quick reference. So it'll list pointers from the rest of the section as well.

**Comment by abhigyantrips** [comment]
Alright, the base content is now complete. Suggestions and nitpicks are welcome.

We'll prevent adding more content, if possible, so that we can merge this page before 2.4 release.

**Comment by abhigyantrips** [comment]
We haven't set up PR deployments yet, so here's a link to see how it'll look in prod. - https://www.abhigyantrips.dev/guide/interactions/modals

**Review by onerandomusername — DISMISSED** [review]

File: guide/docs/interactions/modals.mdx [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+				<DiscordEmbedField fieldTitle="Description">
+					This is the tag's description. It can be as long as 4000 characters.
+				</DiscordEmbedField>
+			</DiscordEmbedFields>
+		</DiscordEmbed>
+	</DiscordMessage>
+</DiscordMessages>
+<br />
+
+This was done using the following code, in the form of a cog.
```
[the remark this answers is not in the document]
Comment (abhigyantrips) [reply guide/docs/interactions/modals.mdx:42]: Already edited this, pushing soon.

**Comment by abhigyantrips** [comment]
We're now ready :3

**Pull request merged**